pkg/apis/configuration/validation/validation.go

```diff
@@ -116,6 +116,36 @@ func validateSize(size string, fieldPath *field.Path) field.ErrorList {
 	return allErrs
 }
 
+func validateBuffer(buff v1alpha1.UpstreamBuffers, fieldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if (v1alpha1.UpstreamBuffers{}) == buff {
+		return allErrs
+	}
+
+	if buff.Number <= 0 {
+		return append(allErrs, field.Invalid(fieldPath, buff.Number, "must be positive"))
+	}
+
+	if buff.Size == "" {
+		return append(allErrs, field.Invalid(fieldPath, buff.Size, "cannot be empty"))
+	}
+
+	if buff.Size == "4k" || buff.Size == "8k" {
```

Comment:
buff.Size is of type size - it can take numbers, like 123 with suffixes m or M and k or K.

I suggest adding a function similar to https://github.com/nginxinc/kubernetes-ingress/blob/master/pkg/apis/configuration/validation/validation.go#L104
See also http://nginx.org/en/docs/syntax.html

Note that it is better to rename the existing function `validateSize` to `validateOffset` and create  a new function `validateSize` that validates using the rules described above.